.github/actions/update-releases-json/index.js

```diff
@@ -0,0 +1,225 @@
+
+    // Patch the minor release date if this needed for RTM releases.
+    if (existingRelease !== undefined && iteration !== undefined) {
+        newRelease.minorReleaseDate = existingRelease.minorReleaseDate;
+    }
+
+    releasesData.releases[releaseMajorMinorVersion] = newRelease;
+
+    // Check if we're going to be putting a version out-of-support.
+    if (minorVersion > 0 && patchVersion === 0 && iteration === undefined) {
+        const endOfSupportDate = new Date(releaseDate.valueOf());
+        endOfSupportDate.setMonth(endOfSupportDate.getMonth() + releasesData.policy.additionalMonthsOfSupportOnNewMinorRelease);
+
+        const previousMinorReleaseKey = `${majorVersion}.${minorVersion-1}`;
+        releasesData.releases[previousMinorReleaseKey].outOfSupportDate = endOfSupportDate;
+        return releasesData.releases[previousMinorReleaseKey];
+    }
+
+    return undefined;
+}
+
+function regenerateCategories(releasesData) {
+    const currentDate = new Date();
+
+    let previewVersions = [];
+    let supportedVersions = [];
+    let unsupportedVersions = [];
+    let versionsToNoLongerMention = [];
+
+    for (const releaseProperty in releasesData.releases) {
+        const release = releasesData.releases[releaseProperty];
+
+        const isRTM = actionUtils.isVersionTagRTM(release.tag);
+        if (!isRTM) {
+            // It's a preview
+            previewVersions.push(releaseProperty);
+        } else if (release.outOfSupportDate === undefined) {
+            supportedVersions.push(releaseProperty);
+        } else {
+            const outOfSupportDate = new Date(release.outOfSupportDate);
+            const dateToNoLongerMention = new Date(release.outOfSupportDate);
+            dateToNoLongerMention.setMonth(dateToNoLongerMention.getMonth() + releasesData.policy.cleanupUnsupportedReleasesAfterMonths);
+
+            if (currentDate >= dateToNoLongerMention) {
+                versionsToNoLongerMention.push(releaseProperty);
+            } else if (currentDate >= outOfSupportDate) {
+                unsupportedVersions.push(releaseProperty);
+            } else {
+                supportedVersions.push(releaseProperty);
+            }
+        }
+    }
+
+    for (const releaseKey of versionsToNoLongerMention) {
+        delete releasesData.releases[releaseKey];
+    }
+
+    // Sort the releases by their major.minor version
+    previewVersions.sort(compareVersionKeys);
+    supportedVersions.sort(compareVersionKeys);
+    unsupportedVersions.sort(compareVersionKeys);
+
+    releasesData.preview = previewVersions;
+    releasesData.supported = supportedVersions;
+    releasesData.unsupported = unsupportedVersions;
+}
+
+// Compares two version keys (major.minor)
+function compareVersionKeys(a, b) {
+    const parsedA = Number(a);
+    const parsedB = Number(b);
+
+    if (isNaN(parsedA) || isNaN(parsedB)) {
+        throw new Error(`Unexpected version keys: ${a} ${b}`)
+    }
+    
+    return parsedB - parsedA;
```

Comment:
nit: Not recommending any action since this likely would never happen with `dotnet-monitor`, but I'm assuming this logic wouldn't work if we had >=10 minor versions?